src/main/java/ch/njol/skript/expressions/ExprSpawner.java

```diff
@@ -0,0 +1,118 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+
+package ch.njol.skript.expressions;
+
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.CreatureSpawner;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.Converter;
+import ch.njol.skript.aliases.Aliases;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityType;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Spawner Type")
+@Description("Retrieves, sets, or resets the spawner's entitytype")
+@Examples({"broadcast \"%spawner's entitytype%\""})
+@Since("INSERT VERSION")
+public class ExprSpawner extends SimplePropertyExpression<Block, EntityType> {
+	
+	static {
+		register(ExprSpawner.class, EntityType.class, "entitytype", "block");
```

Comment:
I did take a look at ExprTypeOf but decided it to be a bit of a hassle, since I didn't want to have to make sure that I didn't screw up prior functionality

Reply:
Returning EntityData instead of EntityType seems to break the entire thing?

Certain spawners properly return their EntityData (e.g. pig, cat, zombie), but certain spawners like cavespider return ``<none>`` for some reason?
```
		for(org.bukkit.entity.EntityType e : org.bukkit.entity.EntityType.values()) {
			CACHE.put(EntityData.parse(e.toString()), e);
		}
```

Similarly, setting spawners' EntityData is also broken. Although it successfully parses, it throws a CCE when it's called for ``ch.njol.skript.entity.EntityType cannot be cast to ch.njol.skript.entity.EntityData`` when using the syntax ``set event-block's entity type to a pig``... any idea?